### ISSUE TITLE
How to add coupons or promotional codes to PaymentSheet?

### ISSUE THREAD

**Issue by NatMuel** [stripe/stripe-react-native#1130]
**Is your feature request related to a problem? Please describe.**
I am using the `paymentIntents.create` with a AWS Amplify GraphQL API to create a paymentIntent as described in the [docs](https://stripe.com/docs/payments/accept-a-payment?platform=react-native&ui=payment-sheet). Afterwards I initialize it with `initPaymentSheet` and subsequently present it using `presentPaymentSheet`. I would like to know how I would be able to add promotional codes linked to coupons/discounts similar to `stripe.checkout.sessions.create` 
```
discounts: [{
    coupon: '{{COUPON_ID}}',
  }]
```
or 
`stripe.checkout.sessions.create` `allow_promotion_codes: true`

**Describe the solution you'd like**
I would like the paymentSheet to be initialized with a promotional code input field where customers can redeem their coupons/discounts.

**Describe alternatives you've considered**
I have considered using the `checkout` and `sessions` API's but found no way for the `paymentIntent` API.

**Comment by lucarducci** [comment]
I'm also looking to how to add `allow_promotion_codes: true` on the `paymentIntent`

**Comment by kvacke** [comment]
Also looking for this. @lucarducci @NatMuel Any progress?

**Comment by wing199901** [comment]
Any updates?

**Comment by NatMuel** [comment]
@kvacke No, unfortunately not. Currently I use a seperate discount input field, handle the discount codes separately and update the payment amount that the paymentInted receives.

**Comment by briancharta** [comment]
x2

**Comment by charliecruzan-stripe** [comment]
Unfortunately this isn't planned yet, you would need to handle this outside of the payment sheet flow (maybe making use of the `customFlow: true` parameter to separate out the "select payment method" and "confirm" steps)

**Comment by jorgerojas26** [comment]
This is a feature that would be awesome to have.

**Comment by astletron** [comment]
+1 for this feature

**Comment by umrashrf** [comment]
Another reason to have this built into the UI is that, I want to show my users that they are getting discounted prices and not the actual price. I want to show them both prices but one with a strikethrough on it.

**Comment by craigRSA** [comment]
How can I redeem a promo code using the Stripe API and ensure Stripe tracks the redemption count, especially when there’s a max number of redemptions?

**Comment by theonetheycallneo** [comment]
still no coupon codes?

**Comment by hamzalakrati** [comment]
+1 for this feature. Customers can add coupons on the stripe billing portal, but not on the PaymentSheet , a must have in my opinion.

**Comment by hamzalakrati** [comment]
@charliecruzan-stripe Is there a way to add this to your roadmap or is it possible for me to contribute to this repo and add that feature if the PR is approved ?

**Comment by craigRSA** [comment]
> @charliecruzan-stripe Is there a way to add this to your roadmap or is it possible for me to contribute to this repo and add that feature if the PR is approved ?

The only way to do this is to lookup the discount % and then apply it yourself to the price so that you can see the was/now. There is a limitation that maximum redemption codes cannot be incremented.

**Comment by hamzalakrati** [comment]
@craigRSA I understand that now this is the only way right now, but quite frankly I don't see why this can't be added to Payment Sheet.

**Comment by damilolasodiq** [comment]
+1 for having this as part of payment intent creation or payment sheet confirmation

**Comment by arioul89** [comment]
+1, clearly a must have

**Comment by Albertbol** [comment]
bump

**Comment by jawad-ul-hassan** [comment]
+1, clearly a must have

**Comment by ABertaud** [comment]
+1, building my own work-around with metadata, but this is truly not optimal

**Comment by Albertbol** [comment]
+1

**Comment by caioedut** [comment]
+1

**Comment by filatovpavel** [comment]
+1

**Comment by ildaneta** [comment]
+1

**Comment by michaelboeding** [comment]
+1

**Comment by lantchou** [comment]
+1

**Comment by 3DTOPO** [comment]
How is it that the StripePaymentSheet still does not have promo code support?

I mean, how hard could it be to add a field connected to promo codes that can be created in the Stripe dashboard? Promo codes are ambiguous on most places I buy from.

**Comment by jonxssc** [comment]
+1

**Comment by hhreszczuk** [comment]
+1. I’m selling a B2C product on both the web and in my own iOS/Android apps. On the web, I can easily support coupon codes through a Checkout Session, which works great. However, there’s no equivalent option for my mobile apps.

I’d prefer not to calculate discounts on my own backend, especially since the discount wouldn’t be reflected on the Stripe receipt. Stripe also already handles coupon validation and restrictions (such as one-time use per customer) very well, so duplicating that logic would be less reliable and create additional maintenance overhead.
